src/status_im/contexts/profile/push_notifications/effects.cljs

```diff
@@ -1,60 +1,130 @@
+    [status-im.contexts.push-notifications.notifications :as pn-notifications]
+    [status-im.contexts.push-notifications.permissions :as pn-permissions]
+    [status-im.contexts.push-notifications.remote-token :as pn-remote-token]
     [utils.re-frame :as rf]))
 
-(def ios-listeners-added? (atom nil))
+(defn attempt-enable-notifications
+  []
+  (-> (pn-permissions/request-notification-permissions)
+      (promesa/then
+       (fn [permissions]
+         (if (:authorized? permissions)
+           permissions
+           (promesa/rejected
+            (ex-info
+             "Failed to authorize push notifications"
+             {:description "Requested and failed to authorize permissions for push notifications"})))))))
 
-(defn enable-ios-notifications
+(defn enable-news-notifications
   []
-  (when-not @ios-listeners-added?
-    (reset! ios-listeners-added? true)
-    (pn-ios/add-listener
-     "register"
-     (fn [token]
-       (rf/dispatch [:push-notifications/registered-for-push-notifications token])))
-    (pn-ios/add-listener
-     "registrationError"
-     (fn [error]
-       (rf/dispatch [:push-notifications/switch-error true error]))))
-  (pn-ios/request-permissions))
+  (-> (attempt-enable-notifications)
+      (promesa/then
+       (fn [_permissions]
+         (pn-remote-token/request-fcm-token {})))
+      (promesa/then
+       (fn [{:keys [fcm-token]}]
+         (logger/log-debug "registered fcm token for status news"
+                           {:fcm-token fcm-token})))
+      (promesa/catch
+        (fn [error]
+          (logger/log-debug "failed to register fcm token for status news"
+                            {:error error})))))
 
-(defn disable-ios-notifications
+(defn disable-news-notifications
+  []
+  (pn-remote-token/revoke-fcm-token {}))
+
+(defn enable-chat-notifications-ios
   []
-  (pn-ios/abandon-permissions)
+  (-> (attempt-enable-notifications)
+      (promesa/then
+       (fn [_permissions]
+         (pn-remote-token/request-apns-token {})))
+      (promesa/then
+       (fn [{:keys [apns-token]}]
+         ;; NOTE(seanstrom): We're only registering iOS devices for remote chat push notifications
+         ;; because Android will use local notifications from the background for chat
+         ;; notifications. This is a temporary workaround that attempts to prevent Android users
+         ;; from receiving two push notifications for each chat message when the app is running
+         ;; the background. This workaround can be removed after we support remote push
+         ;; notifications with message previews that decrypt locally.
+         (when platform/ios?
+           (rf/dispatch [:push-notifications/registered-for-push-notifications apns-token]))))))
+
+(defn disable-chat-notifications-ios
+  []
+  (pn-permissions/release-notification-permissions)
   (rf/dispatch [:push-notifications/unregistered-from-push-notifications]))
 
-(defn enable-android-notifications
+(defn enable-ios-notifications
```

Comment:
We are hiding all platform differences within this namespace, so we can use `defn-` here and in other ios or android-specific functions.

Reply:
Good point, will change 👍